sambacc/textfile.py

```diff
@@ -28,11 +28,13 @@ def read(self) -> None:
         with open(self.path) as f:
             self.readfp(f)
 
-    def write(self) -> None:
-        tpath = self._tmp_path(self.path)
+    def write(self, path: typing.Optional[str] = None) -> None:
```

Comment:
Most of this patch looks OK to me, but this part I find a bit confusing. The path is provided to the object at initialization time, and then again (optionally) here - that seems redundant to me.

Reply:
On systems where this matters, /usr is typically read-only. Any changes to the config files in /usr/etc are written to /etc and prioritized over the ones in /usr/etc. This change allows us to write the contents of the read-in (and modified) file back out to a different location. When the path is unspecified, the existing behavior is preserved.